File: 938.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
struct Solution {
    void dfs(TreeNode *node, const int &low, const int &high, int &res){
        if (node == nullptr){
            return;
        }
        if (low <= node->val && node->val <= high){
            res += node->val;
        }
        if (low < node->val){
            dfs(node->left, low, high, res);
        }
        if (node->val < high){
            dfs(node->right, low, high, res);
        }
    }

    int rangeSumBST(TreeNode* root, int low, int high) {
        int res = 0;
        dfs(root, low, high, res);
        return res;
    }
};

File: 938.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        if root is None:
            return 0
        stack = [root]
        res = 0
        while len(stack) != 0:
            node = stack.pop()
            if low <= node.val <= high:
                res += node.val
            if node.left is not None and low < node.val:
                stack.append(node.left)
            if node.right is not None and node.val < high:
                stack.append(node.right)
        return res
